File: project_euler/025 - 1000-digit Fibonacci number/25.java
import java.math.BigInteger;

class A {

    public static void main(String[] args) {

        BigInteger a = new BigInteger("1");
        BigInteger b = new BigInteger("1");
        BigInteger c = new BigInteger("1");
        BigInteger max = new BigInteger("10");
        max = max.pow(999);

        for (int i = 3; i > 0; i++) {

            c = a.add(b);
            a = b;
            b = c;

            if (c.compareTo(max) < 0) {
                //System.out.println(i+" - "+c);
            } else {
                //System.out.println(i+" - "+c);
                System.out.println("First term in the Fibonacci sequence to contain 1000 digits = "+i);
                break;
            }
        }
    }
}

File: project_euler/025 - 1000-digit Fibonacci number/25.py
ans = 0
num = 0
n=1
a=1
b=1
c=0

while True:
    n+=1
    s = str(b)
    if len(s)==1000:
        ans = n
        num = b
        break
    
    c=a+b
    a=b
    b=c

print "First term in the Fibonacci sequence to contain 1000 digits :", ans
print "Number :", num
